### ISSUE TITLE
Release of Nvim-remote

### ISSUE THREAD

**Issue by mauricekraus** [GustavoKatel/devcontainers-rs#1]
Hey,

I saw your post on Reddit (https://www.reddit.com/r/neovim/comments/qm9lf5/remote_containers_development_in_nvim/).

Do you plan to continue the project, especially now having nvim-remote integrated?

If you give me a good starting point, I'd be happy to contribute.

**Comment by GustavoKatel** [comment]
yes, I plan to continue this project, but I'm waiting a bit on this issue: https://github.com/neovim/neovim/pull/10071

right now this project has the bare minimum and works ok, but if you see the readme in the root you see that there's some missing features, it would be nice to have that completed. Also it's been a while since I checked for new features in the official reference: https://code.visualstudio.com/docs/remote/devcontainerjson-reference

**Comment by sisrfeng** [comment]
I think the title should be release of nvim-remote-containers
see: https://github.com/jamestthompson3/nvim-remote-containers

**Comment by radwo** [comment]
Hi!

Thanks for your previous work on this project!

Lately, https://github.com/neovim/neovim/pull/10071 was merged in https://github.com/neovim/neovim/pull/18375.

Do you still have plans to work on it?

**Comment by GustavoKatel** [comment]
yes, but I don't have a clear timeline to when I can start, I'm busy with another project atm. I'm open to contributions on this tho. happy to guide anyone interested in helping out.

**Comment by c3n21** [comment]
> yes, but I don't have a clear timeline to when I can start, I'm busy with another project atm. I'm open to contributions on this tho. happy to guide anyone interested in helping out.

I would really love to see such feature merged.

My idea would be bind mounting NeoVim's extracted AppImage and use it as a TUI server, so that locally we can run `nvim --remote-ui` to attach to NeoVim running inside the container.

Also this would also mean to bind mount come `XDG_*` directories, but with rootless containers shouldn't be a real issue no?

Also, such configuration I think that shouldn't be saved in project's `.devcontainer.json` but should be something local to the user, for instance a "global" `~/.config/devcontainer-rs/devcontainer.json` that will integrate those configuration in project's configuration, and should only configure NeoVim's server instance (or other editor specific configuration for devcontainers integration) and does the bind mount of XDG_* directories.

**Comment by GustavoKatel** [comment]
tbh I thinks this is already possible. it already supports loading a config from the home folder and merge with the project folder. see: https://github.com/GustavoKatel/devcontainers-rs#how-to

in `2.1 - Starting editor/ide after setting up containers:` try replacing nvim-qt with the TUI

PS.: I haven't tested, so lmk if that works

**Comment by GustavoKatel** [comment]
what's missing is getting the spec reference for `devcontainer.json` up to date with the new features and breaking changes. so bc of that, the container might get a little weird when compared with vscode. not sure how far the diff goes.
